statistics/semantic/src/main/java/com/spotify/heroic/statistics/semantic/SemanticConsumerReporter.java

```diff
@@ -40,6 +41,8 @@
     private final Meter messageIn;
     private final Meter messageError;
     private final Meter consumerSchemaError;
+    private Optional<Long> consumerThreadsWanted;
```

Comment:
It's only set once, before the ConsumerThread threads are started. Tho as you note below, we don't know if the callee will change in the future so the correct way is to ensure it's thread safe. Thanks for the pointer.